### PR TITLE
TMEDIA-398 login block 

### PR DESCRIPTION
**Pull request by mattkim93**
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
_Information about what you changed for this PR_
- adds login block
- it wasn't in scope and I'm cool with pulling it out (we probably need design review), but I added ReCaptcha to the login form. ReCaptcha is turned on from the subs admin, so the UIs need to render a ReCaptcha challenge if Identity config says ReCaptcha is turned on. There are a lot of these kinds of non-mvp flows that I'll start documenting here https://paper.dropbox.com/doc/Subs-Flows--BSHJaow1TyrX0A8zmywXO7KmAg-qVEurOzfPS7tGRtoEPt8e
- I still need to write more tests for the login block before this is ready to merge, but thought I'd throw it up for people to see.
- I'll also be adding the forgot password & sign up link in another block, since it needs to go below 
## Jira Ticket
- [TMEDIA-398](https://arcpublishing.atlassian.net/browse/TMEDIA-398)

## Acceptance Criteria
_copy from ticket_
- User can log in
- Form validation / error state
  - Email will validate format
  - Password should not

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout {branch name}`
2. Run fusion repo with linked blocks `npx fusion start -f -l {blocks to link}`
3. {Next ....}

### Before
_Example: When I clicked the search button, the button turned red._

[include screenshot or gif or link to video, storybook would be awesome]

### After
_Example: When I clicked the search button, the button turned green._
image

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/identity-block/features/login/styles.scss [hunk 1]
```diff
@@ -0,0 +1,36 @@
+.xpmedia-subs-login {
```
Comment (matthewroach) [inline blocks/identity-block/features/login/styles.scss]: Most, if not all of these styles should be controlled via the `HeadlinedSubmitForm` component

Comment (mattkim93) [reply blocks/identity-block/features/login/styles.scss]: ah yep, these are leftover from before i integrated that. I'll remove

File: blocks/identity-block/components/HeadlinedSubmitForm/styles.scss [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 .xpmedia-form-submittable {
   border-bottom: 1px solid $border-rule-color;
-  margin-bottom: map-get($spacers, 'sm');
+  margin-bottom: map-get($spacers, 'md');
```
Comment (JackHowa) [inline blocks/identity-block/components/HeadlinedSubmitForm/styles.scss:11]: were these changes to margin deliberate? I think @nschubach worked on that component? 

```suggestion
  margin-bottom: map-get($spacers, 'sm');
```

File: blocks/identity-block/components/HeadlinedSubmitForm/styles.scss [hunk 3]
```diff
@@ -18,7 +18,7 @@
     }
 
     &:last-of-type {
-      margin-bottom: map-get($spacers, 'sm');
+      margin-bottom: map-get($spacers, 'md');
```
Comment (JackHowa) [inline blocks/identity-block/components/HeadlinedSubmitForm/styles.scss:21]:
```suggestion
      margin-bottom: map-get($spacers, 'sm');
```

File: blocks/identity-block/features/login/default.jsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  const { locale = 'en' } = getProperties(arcSite);
+  const phrases = getTranslatedPhrases(locale);
+
+  const { Identity, isInitialized } = useIdentity();
+
+  const [recaptchaToken, setRecaptchaToken] = useState();
+  const [recaptchaConfig, setRecaptchaConfig] = useState({
+    signinRecaptcha: false,
+    status: 'initial',
+  });
```
Comment (JackHowa) [inline blocks/identity-block/features/login/default.jsx:34]: can we move the recaptcha work to another pr? Looks like we're hitting test coverage limits and removing this would help and the scope
```suggestion
```

File: blocks/identity-block/features/login/default.jsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    redirectURL = document?.referrer;
+  }
+
+  useEffect(() => {
+    const getConfig = async () => {
+      await Identity.getConfig()
+        .then((response) => {
+          const {
+            signinRecaptcha,
+            recaptchaSiteKey,
```
Comment (JackHowa) [inline blocks/identity-block/features/login/default.jsx:48]:
```suggestion
```

File: blocks/identity-block/features/login/default.jsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+          } = response;
+
+          setRecaptchaConfig({
+            signinRecaptcha,
+            recaptchaSiteKey,
+            status: 'success',
+          });
+        })
+        .catch(() => setRecaptchaConfig({ status: 'error' }));
```
Comment (JackHowa) [inline blocks/identity-block/features/login/default.jsx:57]: can we move this recaptcha work to another pr?

File: blocks/identity-block/features/login/default.jsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+    return null;
+  }
+
+  return (
+    <>
+      <HeadlinedSubmitForm
+        headline={phrases.t('identity-block.log-in')}
+        buttonLabel={phrases.t('identity-block.log-in')}
+        onSubmit={({ email, password }) => Identity
+          .login(email, password, { rememberMe: true, recaptchaToken })
```
Comment (JackHowa) [inline blocks/identity-block/features/login/default.jsx:88]: I forget if we made a decision on remember me

File: blocks/identity-block/features/login/default.jsx [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+          type={FIELD_TYPES.EMAIL}
+          validationErrorMessage={phrases.t('identity-block.email-requirements')}
+        />
+        <FormInputField
+          label={phrases.t('identity-block.password')}
+          name="password"
+          required
+          showDefaultError={false}
+          type={FIELD_TYPES.PASSWORD}
+          validationErrorMessage={phrases.t('identity-block.password-requirements')}
```
Comment (JackHowa) [inline blocks/identity-block/features/login/default.jsx:107]: not sure what the standard on password requirements for login. should we even validate this? 

doesn't look like others do 
Screen Shot 2021-09-13 at 15 16 40

Screen Shot 2021-09-13 at 15 17 18

**Review by JackHowa — CHANGES_REQUESTED** [review]
Split out some of this work into a draft pr if you want to work off of that. Not sure if the scope of this ticket has changed 

pr based on some comments: https://github.com/WPMedia/fusion-news-theme-blocks/pull/1082

Not sure if I'm looking at the updated version of this. It doesn't look like it was "ready for review" in the pr initially? 

-------

Would appreciate it if we split out recaptcha work in this. I wasn't able to get the block to show up in the list; I'm guessing I need to reinstall dependencies like recaptcha. 

Also, I noticed there was inline error validation for login that I don't believe we'd need

**Comment by JackHowa** [comment]
also doesn't look like this translation was done yet? 

Screen Shot 2021-09-13 at 15 29 54

**Comment by JackHowa** [comment]
`I'll also be adding the forgot password & sign up link in another block, since it needs to go below` not sure I follow what you mean here?

File: blocks/identity-block/components/HeadlinedSubmitForm/styles.scss (hunk 2 above)
Comment (mattkim93) [reply blocks/identity-block/components/HeadlinedSubmitForm/styles.scss:11]: Yep, at least the zeplin mocks I was looking at had different spacing
https://app.zeplin.io/project/603fa53e2626ed1592e7c0e6/screen/6113e22a4eb86998b18dbed8

File: blocks/identity-block/features/login/default.jsx (hunk 4 above)
Comment (mattkim93) [reply blocks/identity-block/features/login/default.jsx:34]: yep, sounds good. I'll pull it out

File: blocks/identity-block/components/HeadlinedSubmitForm/styles.scss (hunk 2 above)
Comment (nschubach) [reply blocks/identity-block/components/HeadlinedSubmitForm/styles.scss:11]: Bottom margin should be `md` (the space under the border).  I don't know why this was `sm`.
image
So this is correct given the third party buttons are below that.

File: blocks/identity-block/components/HeadlinedSubmitForm/styles.scss (hunk 3 above)
Comment (nschubach) [reply blocks/identity-block/components/HeadlinedSubmitForm/styles.scss:21]: Last input (the confirm password) should be `sm` bottom margin, so this one should not have been changed.
image

File: blocks/identity-block/features/login/default.jsx (hunk 4 above)
Comment (nschubach) [reply blocks/identity-block/features/login/default.jsx:34]: I thought re-captcha was deferred for MVP as well.

File: blocks/identity-block/features/login/default.jsx (hunk 8 above)
Comment (nschubach) [reply blocks/identity-block/features/login/default.jsx:107]: We should _not_ validate.  If we change the rule later, that will lock people out of logging in.

Comment (nschubach) [reply blocks/identity-block/features/login/default.jsx:107]: ~~Thinking this should also apply to the login field too... checking.~~

Email will validate, password should not.

Comment (nschubach) [reply blocks/identity-block/features/login/default.jsx:107]: AC modified

**Review by nschubach — CHANGES_REQUESTED** [review]
Form validation, re-captcha, and margin concerns.

**Comment by matthewroach** [comment]
Using #1082 in favour of this - the code here can be used for reference for reCAPTCHA work in the future
